R/derive_locf_records.R

```diff
@@ -0,0 +1,184 @@
+#' Derive LOCF(Last observation carried forward) records
+#'
+#' Adds LOCF records as new observations for each 'by group' when the data set
+#' does not contain observations for missed visits/time points.
+#'
+#' @param dataset Input dataset
+#'
+#'   The columns specified by the `by_vars` and the `order`
+#'   parameter are expected.
+#'
+#' @param dataset_expected_obs Expected observations dataset
+#'
+#'   Data frame with all the combinations of `PARAMCD`, `PARAM`, `AVISIT`,
+#'   `AVISITN`, ... which are expected in the dataset is expected.
+#'
+#' @param by_vars Grouping variables
+#'
+#'   For each group defined by `by_vars` an observation is added to the output
+#'   dataset. Only variables specified in `by_vars` will be populated
+#'   in the newly created records.
+#'
+#' @param order List of variables for sorting a dataset
+#'
+#'   The dataset is sorted by `order` before carrying the last
+#'   observation (eg. `AVAL`) within each `by_vars` forward.
+#'
+#' @author G Gayatri
+#'
+#' @details For each group (with respect to the variables specified for the
+#'   `by_vars` parameter) an observation is added to the output dataset from
+#'   `dataset_expected_obs`, if the input dataset does not contain observations
+#'   for missed visits/time points or if AVAL is missing.
+#'
+#'   For the new observations, `AVAL` is set to the non-missing `AVAL` of the
+#'   previous observation in the input dataset (when sorted by `order`) and
+#'   `DTYPE` is set to `LOCF`.
+#'
+#' @return The input dataset with the new `LOCF` observations added for each
+#' `by_vars`. Note, a variable will only be populated in the new parameter rows
+#' if it is specified in `by_vars`.
+#'
+#' @keywords der_bds_findings
+#' @family der_bds_findings
```

Comment:
As the function adds records, please use `der_prm_bds_findings` keyword and family.